modules/library/metadata/src/main/java/org/geotools/util/factory/GeoTools.java

```diff
@@ -619,19 +619,33 @@ public static Manifest getManifest(Class<?> type) {
     /**
      * Sets the global {@linkplain LoggerFactory logger factory}.
      *
-     * <p>This method is the same as {@code Logging.GEOTOOLS.setLoggerFactory(factory)}. GeoTools
-     * ships with support for <A
-     * HREF="http://jakarta.apache.org/commons/logging/">Commons-logging</A> and <A
-     * HREF="http://logging.apache.org/log4j/">log4j</A>. This method exists to allow you supply
-     * your own implementation (this is sometimes required when using a GeoTools application in an
+     * <p>This method is the same as calling {@link Logging#setLoggerFactory(factory)} to configure
+     * both {@link Logging#GEOTOOLS} and {@link Logging#JAI} logger creation.
+     *
+     * <p>GeoTools provides logback, log4j, reload4j, and commons-logging factories. This method
+     * exists to allow you supply your own implementation (when using a GeoTools library in an
      * exotic environment like Eclipse, OC4J or your application).
      *
-     * @param factory The logger factory to use.
+     * <p>If {@code null} is used, the Java logging {@linkplain java.util.logging.Formatter
+     * formatter} for console output is replaced by a {@linkplain
+     * org.geotools.util.logging.MonolineFormatter monoline formatter}.
+     *
+     * @param factory The logger factory to use, or null for native java util logging.
      * @see Logging#setLoggerFactory(LoggerFactory)
      * @since 2.4
      */
     public void setLoggerFactory(final LoggerFactory<?> factory) {
```

Comment:
My own feedback is that this method should be ``initLoggerFactory(factory)` as can only really be usefully called during start up.

Trying to use this for GeoWebCache I ended up just talking to ``Logging`` directly as there was no way to check state:
``` java
if(Logging.GEOTOOLS.getLoggerFactory() == null ){
    GeoTools.setLoggerFactory(Log4J2LoggerFactory.getInstance());
}
```

Reply:
oh wow, the method is not even static and cannot be called

Reply:
GeoServer uses ``Logging.ALL.setLoggerFactory(...)``

Reply:
Yes, the key functional difference appears to be using forceMonolineConsoleOutput() when passed in null.
Very old martin decisions, probably to keep test case output tidy.